### PR TITLE
add ab testing for mailcheck

### PR DESCRIPTION
**Pull request by vladikoff**
@dannycoates feedback?

### CODE REVIEW

File: mailcheck/mailcheck-enabled.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  endDate: '2015-11-01',
+  subjectAttributes: ['isMetrics', 'uuid'],
+  independentVariables: ['mailcheckEnabled'],
+  eligibilityFunction: function (subject) {
+    return subject && subject.isMetrics;
+  },
+  groupingFunction: function (subject) {
+    // 50% of participants will see mailcheck
+    return {
+      mailcheckEnabled: subject && this.bernoulliTrial(0.5, subject.sessionId)
```
Comment (shane-tomlinson) [inline mailcheck/mailcheck-enabled.js]: Does `sessionId` need to be `uuid`?

File: mailcheck/mailcheck-enabled.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  name: 'mailcheck is enable or disabled',
```
Comment (shane-tomlinson) [inline mailcheck/mailcheck-enabled.js]: do you want a `d` on the end of `enable`?

**Comment by vladikoff** [comment]
@dannycoates @shane-tomlinson r?

File: defaults.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "avatarLinkVisible": false,
   "communicationPrefsVisible": false,
-  "springCampaign2015": false
+  "springCampaign2015": false,
+  "mailcheckEnabled": false
```
Comment (TDA) [inline defaults.json]: this would look cuter in alpha-order :) 
EDIT: Also a-c-m-s sounds very similar to a particular association we all know ;)

**Comment by shane-tomlinson** [comment]
Can you rebase this @vladikoff?

File: mailcheck/enabled.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  eligibilityFunction: function (subject) {
+    return !! (subject && subject.isMetricsEnabled);
+  },
+  groupingFunction: function (subject) {
+    return {
+      mailcheckEnabled: this.bernoulliTrial(0.5, subject.uuid)
+    };
+  },
+  conclusion: {
+    mailcheckEnabled: false
```
Comment (shane-tomlinson) [inline mailcheck/enabled.js]: @dannycoates - will setting `mailcheckEnabled` to false in `conclusion` force everybody to see `false`?

**Comment by vladikoff** [comment]
@dannycoates could you please also answer this when you get a sec: https://github.com/mozilla/fxa-content-server/pull/2456#discussion_r33943597

File: mailcheck/enabled.js (hunk 4 above)
Comment (dannycoates) [inline mailcheck/enabled.js]: Only after the `endDate`. It isn't necessary to have a `conclusion` block from the get-go though.

Comment (vladikoff) [inline mailcheck/enabled.js]: just to confirm, if we remove `conclusion`, then after `endDate: '2015-11-01',` the value from `defaults.json` will be used, correct?

Comment (dannycoates) [inline mailcheck/enabled.js]: correct

Comment (vladikoff) [inline mailcheck/enabled.js]: Great, thanks!!

**Comment by vladikoff** [comment]
@shane-tomlinson @dannycoates r?

File: mailcheck/enabled.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+module.exports = {
+  name: 'mailcheck is enable or disabled',
+  hypothesis: 'mailcheck will lead to higher confirmation rate of accounts',
+  startDate: '2015-01-01',
+  endDate: '2015-11-01',
+  subjectAttributes: ['uuid', 'isMetricsEnabled', 'forceMailcheck'],
+  independentVariables: ['mailcheckEnabled'],
+  eligibilityFunction: function (subject) {
+    if (subject) {
+      if (subject.forceMailcheck === 'true') {
```
Comment (vladikoff) [inline mailcheck/enabled.js:12]: @dannycoates  I see setting 'true' as a constant doesn't work outside of the module.exports, so I have to keep these here

**Comment by vladikoff** [comment]
Merging this so tests pass on the content server. will tweak if there is more feedback.

File: mailcheck/enabled.js (hunk 5 above)
Comment (dannycoates) [inline mailcheck/enabled.js:12]: yeah, nothing declared outside of these functions will work

**Pull request merged**